In [1]:
import numpy as np
import pandas as pd
# import seaborn as sns
import matplotlib.pyplot as plt
from scipy import stats

### DATA INGESTION

In [2]:
import kagglehub
path = kagglehub.dataset_download("abrambeyer/openintro-possum") 


c:\Users\Adenuga Pelumi\Documents\Complete AI course\week_5\module_new\Lib\site-packages\tqdm\auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm


In [3]:
print("Path to dataset files:", path)

Path to dataset files: C:\Users\Adenuga Pelumi\.cache\kagglehub\datasets\abrambeyer\openintro-possum\versions\1


In [4]:
df= pd.read_csv(r"C:\Users\Adenuga Pelumi\.cache\kagglehub\datasets\abrambeyer\openintro-possum\versions\1\possum.csv")
df.head(3)

case  site  Pop sex  age  hdlngth  skullw  totlngth  taill  footlgth  \
0     1     1  Vic   m  8.0     94.1    60.4      89.0   36.0      74.5   
1     2     1  Vic   f  6.0     92.5    57.6      91.5   36.5      72.5   
2     3     1  Vic   f  6.0     94.0    60.0      95.5   39.0      75.4   

   earconch   eye  chest  belly  
0      54.5  15.2   28.0   36.0  
1      51.2  16.0   28.5   33.0  
2      51.9  15.5   30.0   34.0

In [5]:
df2=df.copy()
df2.head(3)

case  site  Pop sex  age  hdlngth  skullw  totlngth  taill  footlgth  \
0     1     1  Vic   m  8.0     94.1    60.4      89.0   36.0      74.5   
1     2     1  Vic   f  6.0     92.5    57.6      91.5   36.5      72.5   
2     3     1  Vic   f  6.0     94.0    60.0      95.5   39.0      75.4   

   earconch   eye  chest  belly  
0      54.5  15.2   28.0   36.0  
1      51.2  16.0   28.5   33.0  
2      51.9  15.5   30.0   34.0

### PRELIMINARY DATA ANALYSIS

#### Data Inspection and Exploration


In [6]:
df2.tail(4)

case  site    Pop sex  age  hdlngth  skullw  totlngth  taill  footlgth  \
100   101     7  other   m  1.0     88.6    54.7      82.5   39.0      64.4   
101   102     7  other   f  6.0     92.4    55.0      89.0   38.0      63.5   
102   103     7  other   m  4.0     91.5    55.2      82.5   36.5      62.9   
103   104     7  other   f  3.0     93.6    59.9      89.0   40.0      67.6   

     earconch   eye  chest  belly  
100      48.0  14.0   25.0   33.0  
101      45.4  13.0   25.0   30.0  
102      45.9  15.4   25.0   29.0  
103      46.0  14.8   28.5   33.5

In [7]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 104 entries, 0 to 103
Data columns (total 14 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   case      104 non-null    int64  
 1   site      104 non-null    int64  
 2   Pop       104 non-null    object 
 3   sex       104 non-null    object 
 4   age       102 non-null    float64
 5   hdlngth   104 non-null    float64
 6   skullw    104 non-null    float64
 7   totlngth  104 non-null    float64
 8   taill     104 non-null    float64
 9   footlgth  103 non-null    float64
 10  earconch  104 non-null    float64
 11  eye       104 non-null    float64
 12  chest     104 non-null    float64
 13  belly     104 non-null    float64
dtypes: float64(10), int64(2), object(2)
memory usage: 11.5+ KB


In [8]:
df2.describe()

case        site         age     hdlngth      skullw    totlngth  \
count  104.000000  104.000000  102.000000  104.000000  104.000000  104.000000   
mean    52.500000    3.625000    3.833333   92.602885   56.883654   87.088462   
std     30.166206    2.349086    1.909244    3.573349    3.113426    4.310549   
min      1.000000    1.000000    1.000000   82.500000   50.000000   75.000000   
25%     26.750000    1.000000    2.250000   90.675000   54.975000   84.000000   
50%     52.500000    3.000000    3.000000   92.800000   56.350000   88.000000   
75%     78.250000    6.000000    5.000000   94.725000   58.100000   90.000000   
max    104.000000    7.000000    9.000000  103.100000   68.600000   96.500000   

            taill    footlgth    earconch         eye       chest       belly  
count  104.000000  103.000000  104.000000  104.000000  104.000000  104.000000  
mean    37.009615   68.459223   48.130769   15.046154   27.000000   32.586538  
std      1.959518    4.395306    4.109380    1.050374    2.045597    2.761949  
min     32.000000   60.300000   40.300000   12.800000   22.000000   25.000000  
25%     35.875000   64.600000   44.800000   14.400000   25.500000   31.000000  
50%     37.000000   68.000000   46.800000   14.900000   27.000000   32.500000  
75%     38.000000   72.500000   52.000000   15.725000   28.000000   34.125000  
max     43.000000   77.900000   56.200000   17.800000   32.000000   40.000000

In [9]:
print("Location")
print('\n')
print(df['sex'].value_counts())

Location


sex
m    61
f    43
Name: count, dtype: int64


#### Data cleaning


In [10]:
df2.isna().sum()

case        0
site        0
Pop         0
sex         0
age         2
hdlngth     0
skullw      0
totlngth    0
taill       0
footlgth    1
earconch    0
eye         0
chest       0
belly       0
dtype: int64

In [11]:
checking_miss=df2[df2["age"].isna()]
print(checking_miss)

    case  site  Pop sex  age  hdlngth  skullw  totlngth  taill  footlgth  \
43    44     2  Vic   m  NaN     85.1    51.5      76.0   35.5      70.3   
45    46     2  Vic   m  NaN     91.4    54.4      84.0   35.0      72.8   

    earconch   eye  chest  belly  
43      52.6  14.4   23.0   27.0  
45      51.2  14.4   24.5   35.0  


In [12]:
df2['age']=df2['age'].fillna(df2['age'].median())


In [13]:
df2.loc[45, :]

case          46
site           2
Pop          Vic
sex            m
age          3.0
hdlngth     91.4
skullw      54.4
totlngth    84.0
taill       35.0
footlgth    72.8
earconch    51.2
eye         14.4
chest       24.5
belly       35.0
Name: 45, dtype: object

In [14]:
df2.isna().sum()

case        0
site        0
Pop         0
sex         0
age         0
hdlngth     0
skullw      0
totlngth    0
taill       0
footlgth    1
earconch    0
eye         0
chest       0
belly       0
dtype: int64

In [15]:
# we still have footlgth to fill
df2["footlgth"]=df2["footlgth"].fillna(df2["footlgth"].median())
df2

case  site    Pop sex  age  hdlngth  skullw  totlngth  taill  footlgth  \
0       1     1    Vic   m  8.0     94.1    60.4      89.0   36.0      74.5   
1       2     1    Vic   f  6.0     92.5    57.6      91.5   36.5      72.5   
2       3     1    Vic   f  6.0     94.0    60.0      95.5   39.0      75.4   
3       4     1    Vic   f  6.0     93.2    57.1      92.0   38.0      76.1   
4       5     1    Vic   f  2.0     91.5    56.3      85.5   36.0      71.0   
..    ...   ...    ...  ..  ...      ...     ...       ...    ...       ...   
99    100     7  other   m  1.0     89.5    56.0      81.5   36.5      66.0   
100   101     7  other   m  1.0     88.6    54.7      82.5   39.0      64.4   
101   102     7  other   f  6.0     92.4    55.0      89.0   38.0      63.5   
102   103     7  other   m  4.0     91.5    55.2      82.5   36.5      62.9   
103   104     7  other   f  3.0     93.6    59.9      89.0   40.0      67.6   

     earconch   eye  chest  belly  
0        54.5  15.2   28.0   36.0  
1        51.2  16.0   28.5   33.0  
2        51.9  15.5   30.0   34.0  
3        52.2  15.2   28.0   34.0  
4        53.2  15.1   28.5   33.0  
..        ...   ...    ...    ...  
99       46.8  14.8   23.0   27.0  
100      48.0  14.0   25.0   33.0  
101      45.4  13.0   25.0   30.0  
102      45.9  15.4   25.0   29.0  
103      46.0  14.8   28.5   33.5  

[104 rows x 14 columns]

In [16]:
df2.head(5)

case  site  Pop sex  age  hdlngth  skullw  totlngth  taill  footlgth  \
0     1     1  Vic   m  8.0     94.1    60.4      89.0   36.0      74.5   
1     2     1  Vic   f  6.0     92.5    57.6      91.5   36.5      72.5   
2     3     1  Vic   f  6.0     94.0    60.0      95.5   39.0      75.4   
3     4     1  Vic   f  6.0     93.2    57.1      92.0   38.0      76.1   
4     5     1  Vic   f  2.0     91.5    56.3      85.5   36.0      71.0   

   earconch   eye  chest  belly  
0      54.5  15.2   28.0   36.0  
1      51.2  16.0   28.5   33.0  
2      51.9  15.5   30.0   34.0  
3      52.2  15.2   28.0   34.0  
4      53.2  15.1   28.5   33.0

In [17]:
df2.isna().sum()

case        0
site        0
Pop         0
sex         0
age         0
hdlngth     0
skullw      0
totlngth    0
taill       0
footlgth    0
earconch    0
eye         0
chest       0
belly       0
dtype: int64

In [18]:
# No missing data, lets check for duplicatee
df2.duplicated().sum()

np.int64(0)

### Statistical Analysis
#### Measure of central Tendency

In [19]:
# We have to deal with columns that are not figures
df2.head()

case  site  Pop sex  age  hdlngth  skullw  totlngth  taill  footlgth  \
0     1     1  Vic   m  8.0     94.1    60.4      89.0   36.0      74.5   
1     2     1  Vic   f  6.0     92.5    57.6      91.5   36.5      72.5   
2     3     1  Vic   f  6.0     94.0    60.0      95.5   39.0      75.4   
3     4     1  Vic   f  6.0     93.2    57.1      92.0   38.0      76.1   
4     5     1  Vic   f  2.0     91.5    56.3      85.5   36.0      71.0   

   earconch   eye  chest  belly  
0      54.5  15.2   28.0   36.0  
1      51.2  16.0   28.5   33.0  
2      51.9  15.5   30.0   34.0  
3      52.2  15.2   28.0   34.0  
4      53.2  15.1   28.5   33.0

In [20]:
# dealing with data types
df2.dtypes

case          int64
site          int64
Pop          object
sex          object
age         float64
hdlngth     float64
skullw      float64
totlngth    float64
taill       float64
footlgth    float64
earconch    float64
eye         float64
chest       float64
belly       float64
dtype: object

In [22]:
# setting non alpha columns
Numerical_columns=[col for col in df2.select_dtypes(include=['int64', 'float64'])]
alpha_columns=[col for col in df2.select_dtypes(include=['object', 'category'])]

In [23]:
Numerical_columns

['case',
 'site',
 'age',
 'hdlngth',
 'skullw',
 'totlngth',
 'taill',
 'footlgth',
 'earconch',
 'eye',
 'chest',
 'belly']

In [29]:
print("Location")
print(df2['Pop'].value_counts())
print('\n')
print(df2['sex'].value_counts())

Location
Pop
other    58
Vic      46
Name: count, dtype: int64


sex
m    61
f    43
Name: count, dtype: int64


In [24]:
df2.set_index('case')

site    Pop sex  age  hdlngth  skullw  totlngth  taill  footlgth  \
case                                                                     
1        1    Vic   m  8.0     94.1    60.4      89.0   36.0      74.5   
2        1    Vic   f  6.0     92.5    57.6      91.5   36.5      72.5   
3        1    Vic   f  6.0     94.0    60.0      95.5   39.0      75.4   
4        1    Vic   f  6.0     93.2    57.1      92.0   38.0      76.1   
5        1    Vic   f  2.0     91.5    56.3      85.5   36.0      71.0   
...    ...    ...  ..  ...      ...     ...       ...    ...       ...   
100      7  other   m  1.0     89.5    56.0      81.5   36.5      66.0   
101      7  other   m  1.0     88.6    54.7      82.5   39.0      64.4   
102      7  other   f  6.0     92.4    55.0      89.0   38.0      63.5   
103      7  other   m  4.0     91.5    55.2      82.5   36.5      62.9   
104      7  other   f  3.0     93.6    59.9      89.0   40.0      67.6   

      earconch   eye  chest  belly  
case                                
1         54.5  15.2   28.0   36.0  
2         51.2  16.0   28.5   33.0  
3         51.9  15.5   30.0   34.0  
4         52.2  15.2   28.0   34.0  
5         53.2  15.1   28.5   33.0  
...        ...   ...    ...    ...  
100       46.8  14.8   23.0   27.0  
101       48.0  14.0   25.0   33.0  
102       45.4  13.0   25.0   30.0  
103       45.9  15.4   25.0   29.0  
104       46.0  14.8   28.5   33.5  

[104 rows x 13 columns]

In [33]:
just_number=df2.iloc[0:, 4:]
just_number

age  hdlngth  skullw  totlngth  taill  footlgth  earconch   eye  chest  \
0    8.0     94.1    60.4      89.0   36.0      74.5      54.5  15.2   28.0   
1    6.0     92.5    57.6      91.5   36.5      72.5      51.2  16.0   28.5   
2    6.0     94.0    60.0      95.5   39.0      75.4      51.9  15.5   30.0   
3    6.0     93.2    57.1      92.0   38.0      76.1      52.2  15.2   28.0   
4    2.0     91.5    56.3      85.5   36.0      71.0      53.2  15.1   28.5   
..   ...      ...     ...       ...    ...       ...       ...   ...    ...   
99   1.0     89.5    56.0      81.5   36.5      66.0      46.8  14.8   23.0   
100  1.0     88.6    54.7      82.5   39.0      64.4      48.0  14.0   25.0   
101  6.0     92.4    55.0      89.0   38.0      63.5      45.4  13.0   25.0   
102  4.0     91.5    55.2      82.5   36.5      62.9      45.9  15.4   25.0   
103  3.0     93.6    59.9      89.0   40.0      67.6      46.0  14.8   28.5   

     belly  
0     36.0  
1     33.0  
2     34.0  
3     34.0  
4     33.0  
..     ...  
99    27.0  
100   33.0  
101   30.0  
102   29.0  
103   33.5  

[104 rows x 10 columns]

In [49]:

for col in just_number.columns:
    min_value = df2[col].min()
    max_value = df2[col].max()
    mean = df2[col].mean()
    median = df2[col].median()
    mode = df2[col].mode().iloc[0]
    value_range = max_value - min_value
    print(f"For {col}")
    print(f"             Maximum value: {max_value}")
    print(f"             Minimum value: {min_value}")
    print(f"             Mean: {mean:.1f}")
    print(f"             Median: {median:.1f}")
    print(f"             Mode: {mode}")
    print(f"             Range: {value_range}")
    print()

For age
             Maximum value: 9.0
             Minimum value: 1.0
             Mean: 3.8
             Median: 3.0
             Mode: 3.0
             Range: 8.0

For hdlngth
             Maximum value: 103.1
             Minimum value: 82.5
             Mean: 92.6
             Median: 92.8
             Mode: 93.3
             Range: 20.599999999999994

For skullw
             Maximum value: 68.6
             Minimum value: 50.0
             Mean: 56.9
             Median: 56.3
             Mode: 57.6
             Range: 18.599999999999994

For totlngth
             Maximum value: 96.5
             Minimum value: 75.0
             Mean: 87.1
             Median: 88.0
             Mode: 89.0
             Range: 21.5

For taill
             Maximum value: 43.0
             Minimum value: 32.0
             Mean: 37.0
             Median: 37.0
             Mode: 38.0
             Range: 11.0

For footlgth
             Maximum value: 77.9
             Minimum value: 60.3
             M

#### General Interpretation

1)Age

Range: 1–9 years, mean = 3.8 years.

The distribution is skewed toward younger ages since the median (3.0) and mode (3.0) are lower than the mean (3.8).

Suggests most individuals are young, but a few older ones raise the mean.

2)Head Length (hdlngth)

Range: 82.5–103.1 mm, mean ≈ 92.6 mm.

Median and mode are close to the mean, indicating a fairly symmetric distribution.

Variation is moderate (range ≈ 20.6 mm).

3)Skull Width (skullw)

Range: 50.0–68.6 mm, mean = 56.9 mm.

Median (56.3) and mode (57.6) are close to the mean → balanced distribution.

Indicates consistent skull proportions among individuals.

4)Total Length (totlngth)

Range: 75–96.5 cm, mean = 87.1 cm.

Median (88.0) and mode (89.0) slightly higher than mean → slight left skew.

Shows a fairly uniform body size.

5)Tail Length (taill)

Range: 32–43 cm, mean = 37 cm.

Mean, median, and mode are nearly the same → symmetric distribution.

Suggests very consistent tail sizes.

6)Foot Length (footlgth)

Range: 60.3–77.9 mm, mean = 68.5 mm.

Median (68.0) and mode (73.2) are slightly different, suggesting mild variation.

Range (≈ 17.6 mm) shows some diversity in foot size.

7)Ear Conch (earconch)

Range: 40.3–56.2 mm, mean = 48.1 mm.

Median (46.8) < mean, mode (44.9) < mean → right-skewed distribution.

A few individuals with larger ear sizes increase the mean.

8)Eye Diameter (eye)

Range: 12.8–17.8 mm, mean = 15.0 mm.

Median (14.9) and mode (14.4) close → balanced.

Very small range (5 mm) → highly consistent eye sizes.

9)Chest

Range: 22–32 cm, mean = 27 cm.

Mean, median, and mode overlap → symmetrical.

Indicates a stable chest size across individuals.

10)Belly

Range: 25–40 cm, mean ≈ 32.6 cm.

Median (32.5) and mode (32.0) align with mean → symmetry.

Some variation, but overall balanced body shape.

#### Key Observations

Consistency: Traits like tail length, chest, eye size, and belly show strong symmetry (mean ≈ median ≈ mode). This suggests these measurements are stable and less influenced by outliers.

Variation: Age, foot length, and ear conch show noticeable differences between mean, median, and mode, suggesting skewness or more variation in those traits.

Biological implication (if this is for animals):

Younger individuals dominate the sample (most are around 3 years).

Core body proportions (head, skull, chest, belly, tail) are stable, which could indicate developmental uniformity.

Peripheral traits (ear, foot) show more variation, possibly due to environmental or genetic factors.

In [50]:
df2.head()

case  site  Pop sex  age  hdlngth  skullw  totlngth  taill  footlgth  \
0     1     1  Vic   m  8.0     94.1    60.4      89.0   36.0      74.5   
1     2     1  Vic   f  6.0     92.5    57.6      91.5   36.5      72.5   
2     3     1  Vic   f  6.0     94.0    60.0      95.5   39.0      75.4   
3     4     1  Vic   f  6.0     93.2    57.1      92.0   38.0      76.1   
4     5     1  Vic   f  2.0     91.5    56.3      85.5   36.0      71.0   

   earconch   eye  chest  belly  
0      54.5  15.2   28.0   36.0  
1      51.2  16.0   28.5   33.0  
2      51.9  15.5   30.0   34.0  
3      52.2  15.2   28.0   34.0  
4      53.2  15.1   28.5   33.0

In [56]:
# Analyzing According to gender
# let's do for male
df_male= df2[df2['sex']== 'm']
df_male

case  site    Pop sex  age  hdlngth  skullw  totlngth  taill  footlgth  \
0       1     1    Vic   m  8.0     94.1    60.4      89.0   36.0      74.5   
6       7     1    Vic   m  2.0     95.3    58.2      89.5   36.0      71.5   
12     13     1    Vic   m  5.0     95.1    59.9      89.5   36.0      71.0   
13     14     1    Vic   m  3.0     95.4    57.6      91.5   36.0      74.3   
14     15     1    Vic   m  5.0     92.9    57.6      85.5   34.0      69.7   
..    ...   ...    ...  ..  ...      ...     ...       ...    ...       ...   
96     97     7  other   m  3.0     91.0    53.1      86.0   38.0      63.8   
97     98     7  other   m  5.0     93.2    68.6      84.0   35.0      65.6   
99    100     7  other   m  1.0     89.5    56.0      81.5   36.5      66.0   
100   101     7  other   m  1.0     88.6    54.7      82.5   39.0      64.4   
102   103     7  other   m  4.0     91.5    55.2      82.5   36.5      62.9   

     earconch   eye  chest  belly  
0        54.5  15.2   28.0   36.0  
6        52.0  14.2   30.0   34.5  
12       49.8  15.8   27.0   32.0  
13       53.7  15.1   28.0   31.5  
14       51.8  15.7   28.0   35.0  
..        ...   ...    ...    ...  
96       46.0  14.5   25.0   31.5  
97       44.3  14.5   28.5   32.0  
99       46.8  14.8   23.0   27.0  
100      48.0  14.0   25.0   33.0  
102      45.9  15.4   25.0   29.0  

[61 rows x 14 columns]

In [60]:
df_male_adj=df_male.iloc[0:, 4:]
df_male_adj

age  hdlngth  skullw  totlngth  taill  footlgth  earconch   eye  chest  \
0    8.0     94.1    60.4      89.0   36.0      74.5      54.5  15.2   28.0   
6    2.0     95.3    58.2      89.5   36.0      71.5      52.0  14.2   30.0   
12   5.0     95.1    59.9      89.5   36.0      71.0      49.8  15.8   27.0   
13   3.0     95.4    57.6      91.5   36.0      74.3      53.7  15.1   28.0   
14   5.0     92.9    57.6      85.5   34.0      69.7      51.8  15.7   28.0   
..   ...      ...     ...       ...    ...       ...       ...   ...    ...   
96   3.0     91.0    53.1      86.0   38.0      63.8      46.0  14.5   25.0   
97   5.0     93.2    68.6      84.0   35.0      65.6      44.3  14.5   28.5   
99   1.0     89.5    56.0      81.5   36.5      66.0      46.8  14.8   23.0   
100  1.0     88.6    54.7      82.5   39.0      64.4      48.0  14.0   25.0   
102  4.0     91.5    55.2      82.5   36.5      62.9      45.9  15.4   25.0   

     belly  
0     36.0  
6     34.5  
12    32.0  
13    31.5  
14    35.0  
..     ...  
96    31.5  
97    32.0  
99    27.0  
100   33.0  
102   29.0  

[61 rows x 10 columns]

In [62]:

for col in df_male_adj.columns:
    min_value = df2[col].min()
    max_value = df2[col].max()
    mean = df2[col].mean()
    median = df2[col].median()
    mode = df2[col].mode().iloc[0]
    value_range = max_value - min_value
    print("Measure of center for Only Male ")
    print(f"For {col}")
    print(f"             Maximum value: {max_value}")
    print(f"             Minimum value: {min_value}")
    print(f"             Mean: {mean:.1f}")
    print(f"             Median: {median:.1f}")
    print(f"             Mode: {mode}")
    print(f"             Range: {value_range}")
    print()

Measure of center for Only Male 
For age
             Maximum value: 9.0
             Minimum value: 1.0
             Mean: 3.8
             Median: 3.0
             Mode: 3.0
             Range: 8.0

Measure of center for Only Male 
For hdlngth
             Maximum value: 103.1
             Minimum value: 82.5
             Mean: 92.6
             Median: 92.8
             Mode: 93.3
             Range: 20.599999999999994

Measure of center for Only Male 
For skullw
             Maximum value: 68.6
             Minimum value: 50.0
             Mean: 56.9
             Median: 56.3
             Mode: 57.6
             Range: 18.599999999999994

Measure of center for Only Male 
For totlngth
             Maximum value: 96.5
             Minimum value: 75.0
             Mean: 87.1
             Median: 88.0
             Mode: 89.0
             Range: 21.5

Measure of center for Only Male 
For taill
             Maximum value: 43.0
             Minimum value: 32.0
             Mean: 37.0
    

In [66]:
for col in just_number:
    Variance_d = df2[col].var()
    standard_d = df2[col].std()
    kurtosis= df2[col].kurtosis()
    q1 = np.quantile(df2[col], 0.25)
    q2 = np.quantile(df2[col], 0.50)  # median
    q3 = np.quantile(df2[col], 0.75)
    IQR = q3 - q1

    print(f" The variance of {col} is {Variance_d:.2f}")
    print(f" The standard deviation of {col} is {standard_d:.2f}")
    print(f" Kurtosis is: {kurtosis:.2f}")
    print(f" The interquartile range of {col} is {IQR}")
    print()



 The variance of age is 3.59
 The standard deviation of age is 1.89
 Kurtosis is: -0.23
 The interquartile range of age is 2.25

 The variance of hdlngth is 12.77
 The standard deviation of hdlngth is 3.57
 Kurtosis is: 0.89
 The interquartile range of hdlngth is 4.049999999999997

 The variance of skullw is 9.69
 The standard deviation of skullw is 3.11
 Kurtosis is: 2.48
 The interquartile range of skullw is 3.125

 The variance of totlngth is 18.58
 The standard deviation of totlngth is 4.31
 Kurtosis is: -0.10
 The interquartile range of totlngth is 6.0

 The variance of taill is 3.84
 The standard deviation of taill is 1.96
 Kurtosis is: 0.40
 The interquartile range of taill is 2.125

 The variance of footlgth is 19.13
 The standard deviation of footlgth is 4.37
 Kurtosis is: -1.18
 The interquartile range of footlgth is 7.849999999999994

 The variance of earconch is 16.89
 The standard deviation of earconch is 4.11
 Kurtosis is: -1.36
 The interquartile range of earconch is 7.2